### PR TITLE
Added README.rst file

### LINKED ISSUE #8: Update README file
Add short description, build status badge, how to setup and run tests.

### PR DESCRIPTION
**Pull request by anurag-ks**
Implements and closes #8

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+How to Setup ?
+================
+Install your cloned fork:
+
+.. code-block:: shell
+
+    git clone https://github.com/openwisp/django-ipam.git
+    cd django-ipam
+    pip install -r requirements.txt
+    python setup.py develop
```
Comment (rohithasrk) [inline README.rst]: Doesn't `python setup.py develop` install requirements as well?

Comment (anurag-ks) [reply README.rst]: My bad, I forgot about that. Thank you for pointing it out :+1:

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+.. image:: http://netjsonconfig.openwisp.org/en/latest/_images/openwisp.org.svg
+  :target: http://openwisp.org
+
+**django-ipam** is Django reusable app for IP Address Management.
+
+------------
+
+How to Setup ?
+================
```
Comment (nemesifier) [inline README.rst]: the number of  `=` should be equal to the number of the letters above, eg:

```
How to Setup?
=============
```

Also, please don't leave any space between `Setup` and `?`

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+.. code-block:: shell
+
+    git clone https://github.com/openwisp/django-ipam.git
+    cd django-ipam
+    python setup.py develop
+
+------------
+
+Run Tests
+================
```
Comment (nemesifier) [inline README.rst]: same here

**Review by nemesifier — CHANGES_REQUESTED** [review]
Not enough info, I think you copied the one of django-freeradius, which is hosting the documentation on readthedocs and therefore has a very small and lean README.

For this project we can host the docs directly here on github in its README so it will need to be more complete, see two good projects that take a similar approach here:

- https://github.com/openwisp/django-netjsonconfig
- https://github.com/openwisp/django-x509

Copy the structure and leave the parts related to settings empty, regarding the setup, write the least amount of stuff to get it working. Remember: people will try this stuff and if it won't work they'll get stuck and won't be able to use it.

**Comment by anurag-ks** [comment]
@nemesisdesign I have added some more stuff, but as of now since we don't have much features to document it still looks kinda empty. I will obviously continue updating the README as we progress.

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+* RESTful API to for CRUD operations
+* Possibility to search for an IP or subnet
+* CSV Import and Export of subnets and their IPs
+
+------------
+
+Project Goals
+=============
+
+* provide a django reusable app with features of IP Address management.
```
Comment (nemesifier) [inline README.rst]: remove final `.`

File: README.rst [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+* provide abstract models which can be extended into other django based apps.
```
Comment (nemesifier) [inline README.rst]: remove final `.`

File: README.rst [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+------------
+
+Dependencies
+============
+
+.. code-block:: shell
+
+    Python 3.x
```
Comment (nemesifier) [inline README.rst]: put `Python 3.4 or greather`

File: README.rst [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+Install Development Version
+===========================
+Install the development version using the following commands -
```
Comment (nemesifier) [inline README.rst]: leave blank line above, replace the final ` -` with `:`

File: README.rst [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+    git clone https://github.com/openwisp/django-ipam.git
+    cd django-ipam
+    python setup.py develop
+
+------------
+
+Setup (Integrate into other Django Apps)
+========================================
+Add ``django_ipam`` to ``INSTALLED_APPS``:
```
Comment (nemesifier) [inline README.rst]: Leave blank line above

File: README.rst [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+    ./manage.py migrate
+
+------------
+
+Run Tests
+=========
+
+.. code-block:: shell
+
+    python runtests.py
```
Comment (nemesifier) [inline README.rst]: use `./runtests.py` for consistency with other openwisp modules, ensure it's executable (`chmod +x runtests.py`)

File: README.rst [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+------------
+
+.. contents:: **Table of Contents**:
+   :backlinks: none
+   :depth: 3
+
+------------
+
+Current Features
```
Comment (nemesifier) [inline README.rst]: Replace this with `TODO: features we are working on during GSoC 18`, when GSoC will be over we'll update it

**Review by nemesifier — CHANGES_REQUESTED** [review]
last round of improvements and then this is done :+1:

**Review by nemesifier — APPROVED** [review]
:+1: go ahead and push to master

**Pull request merged**